site/src/pages/release-notes/0.99.8.mdx

```diff
@@ -0,0 +1,188 @@
+  CompletableFuture<HttpFileAttributes> attrs =
+      httpFile.readAttributes(CommonPools.blockingTaskExecutor());
+  CompletableFuture<ResponseHeaders> headers =
+      httpFile.readHeaders(CommonPools.blockingTaskExecutor());
+  ```
+- You can now easily customize <type://MeterIdPrefixFunction> using
+  <type://MeterIdPrefixFunction#andThen(MeterIdPrefixFunctionCustomizer)>. #2839
+  ```java
+  MeterIdPrefixFunction
+          .ofDefault("hoge")
+          .andThen((registry, log, id) -> {
+              return id.withTags("grpc-status", log.responseTrailers().get("grpc-status"));
+          });
+  ```
+- Both synchronous and asynchronous APIs to select an <type://Endpoint> are added to
+  <type://EndpointSelector> and thus <type://EndpointGroup> which is a sub-interface. #1910 #2837
+  ```java
+  ClientRequestContext ctx = ...
+  EndpointGroup group = ...
+   // synchronous selection!
+  Endpoint endpoint = group.selectNow(ctx);
+   // asynchronous selection!
+  CompletableFuture<Endpoint> future = group.select(ctx, ctx.eventLoop(), 1000 /* timeout */ );
+  ```
+  - Thanks to this change, you can now avoid the <type://EmptyEndpointGroupException> at least
+    for the case where the <typeplural://Endpoint> disappear temporarily.
+- You can now get the snapshot status of circuit breakers from monitoring system. #2841
+- You can now schedule a request timeout in nanoseconds. #2864
+  - <type://ServiceRequestContext#setRequestTimeout(TimeoutMode,Duration)>
+  - <type://ClientRequestContext#setResponseTimeout(TimeoutMode,Duration)>
+
+## 📈 Improvements
+
+- You can now see the error message by `ResourceLeakDetector` if the `DefaultClientFactory` is not closed
+  properly. #2847
+- The initial connection latency is lowered by giving up hopeless HTTP/1.1 connection early when
+  HTTP/2 upgrade request fails. #2852
+
+## 🛠️ Bug fixes
+
+- Armeria server does not convert the `Host` header to `:authority` header anymore. #2846 #2850
+  - <type://RequestHeaders#authority()> looks into `Host` header if `:authority` header is missing.
+- You can retry when <type://DnsTimeoutException> is raised using <type://RetryRule#onUnprocessed()>. #2836
+- You no longer see `Http2Exception: failed to consume a HEADE` error. #2832
+- You no longer see `You tried to set the content preview twice` warning unless
+  you really did apply content previewing decorator twice. #2830 #2832
+- A non-base <type://WebClient> respects the user defined scheme correctly. #2835
+- The specified port when building <type://EurekaUpdatingListener> and <type://ZooKeeperUpdatingListener>
+  is used in the registration information of an Armeria server. #2826
+
+## 🏚️ Deprecations
+
+- All Spring registration beans are deprecated. Use <type://ArmeriaServerConfigurator> and
+  <type://DocServiceConfigurator> instead. #2787 #2838
+  - Deprecated beans: `HttpServiceRegistrationBean`, `AnnotatedServiceRegistrationBean`,
+    `GrpcServiceRegistrationBean` and `ThriftServiceRegistrationBean`
```

Comment:
`MeterIdPrefixFunctionFactory` has been deprecated in favor of `RequestLog.serviceName()`. #2780 #2820